Bio/SeqIO/FastaIO.py

```diff
@@ -104,34 +115,45 @@ def FastaTwoLineParser(handle):
     ValueError: Expected FASTA record starting with '>' character. Perhaps this file is using FASTA line wrapping? Got: 'MTFGLVYTVYATAIDPKKGSLGTIAPIAIGFIVGANI'
 
     """
+    try:
+        handle = open(source)
+    except TypeError:
+        handle = source
+        if handle.read(0) != "":
+            raise ValueError("Fasta files must be opened in text mode") from None
+
     idx = -1  # for empty file
-    for idx, line in enumerate(handle):
-        if idx % 2 == 0:  # title line
-            if line[0] != ">":
-                raise ValueError(
-                    "Expected FASTA record starting with '>' character. "
-                    "Perhaps this file is using FASTA line wrapping? "
-                    "Got: '{}'".format(line)
-                )
-            title = line[1:].rstrip()
-        else:  # sequence line
-            if line[0] == ">":
-                raise ValueError(
-                    "Two '>' FASTA lines in a row. Missing sequence line "
-                    "if this is strict two-line-per-record FASTA format. "
-                    "Have '>{}' and '{}'".format(title, line)
-                )
-            yield title, line.strip()
-
-    if idx == -1:
-        pass  # empty file
-    elif idx % 2 == 0:  # on a title line
-        raise ValueError(
-            "Missing sequence line at end of file if this is strict "
-            "two-line-per-record FASTA format. Have title line '{}'".format(line)
+    try:
+        for idx, line in enumerate(handle):
+            if idx % 2 == 0:  # title line
+                if not line.startswith(">"):
+                    raise ValueError(
+                        "Expected FASTA record starting with '>' character. "
+                        "Perhaps this file is using FASTA line wrapping? "
+                        "Got: '{}'".format(line)
+                    )
+                title = line[1:].rstrip()
+            else:  # sequence line
+                if line.startswith(">"):
+                    raise ValueError(
+                        "Two '>' FASTA lines in a row. Missing sequence line "
+                        "if this is strict two-line-per-record FASTA format. "
+                        "Have '>{}' and '{}'".format(title, line)
+                    )
+                yield title, line.strip()
+
+        if idx == -1:
+            pass  # empty file
+        elif idx % 2 == 0:  # on a title line
+            raise ValueError(
+                "Missing sequence line at end of file if this is strict "
+                "two-line-per-record FASTA format. Have title line '{}'".format(line)
         )
```

Comment:
This bracket also needs the indentation to the right, that's what black didn't like on line 151 column 9:

```
$ flake8
./Bio/SeqIO/FastaIO.py:151:9: BLK100 Black would make changes.
...
```

Note black might make more changes - the flake8 plugin reports only the first (otherwise it gets too noisy).